Towards Adding UL Samples

JHiltbrand opened the pull request:
--This PR includes changes to add a TTToHadronic sample for each of the four years 16, 16APV, 17, and 18 to exemplify the workflow for adding samples in order to get feedback that we are doing things correctly.
--The PR will remain a draft as we go along.
--JEC/JER/PU file paths have simply been copied from previous version of 16, 17, and 18 eras

Commit:
begin adding samples for UL

Commit:
forgot to add 16 folder

semrat committed:
added cff dictionaries to get the root files corresponding to each sample for each year

semrat committed:
added the file lists which include the root files corresponding to each sample for each year

semrat committed:
added the condor submit dictionaries

semrat committed:
fixed the empty dictionary for Summer20UL16APV

semrat committed:
fixed the empty dictionary for Summer20UL16APV

semrat committed:
fixed the other empty dictionary for Summer20UL16APV

semrat committed:
updated the sample lists after fixed the empty cff dictionary issue

aperloff commented on Production/test/condorSub/dict_Summer20UL16APV_tt.py:
```diff
@@ -0,0 +1,6 @@
+flist = {
+    "scenario": "Summer20UL16APV",
+    "samples": [
+        ['Summer20UL16APV.TTToHadronic_TuneCP5_13TeV-powheg-pythia8'],
```
Why did TTToHadronic get put in this file when the other TTJets samples got put into the ttbar file?

JHiltbrand replied:
I think this was just a disconnect between when I made "example" changes to include a new sample and when @semrat started adding more samples in. We will of course consolidate and clean this up.

aperloff replied:
This is still not consolidated. Please do clean this up.

aperloff commented on Production/test/condorSub/ul_condorSubmit_dictionaries/dict_Summer20UL17_singlet.py:
```diff
@@ -0,0 +1,13 @@
+   "scenario": "Summer20UL17",
+   "samples": [
+
+       ['Summer20UL17.ST_t-channel_eleDecays_TuneCP5_13TeV-comphep-pythia8'],
```
There were a lot more single top file lists added. Either some are missing here or you don't need all of the file lists you added.

semrat replied:
Actually, I recently realized that some of samples are missing !!!

- for example 2018 qcd condor submit dictionary is empty, 
- or if you compare the each year to each other, the condor submit dictionaries don't have the same amount of samples in it
- And I don't know why, I have to check !

aperloff commented on Production/python/scenarios.py:
```diff
@@ -108,6 +108,30 @@ def __init__(self,sname):
+                tagname="PAT",
+                hlttagname="HLT",
+                geninfo=True,
+                jecfile=self.absdatapath+"/jec/Summer16_07Aug2017_V11_MC",
```
Have the JEC and JER files been updated? These seem like the old ones.

JHiltbrand replied:
When I initialized the new scenarios I just copied over the old paths for these. @binayab is currently working on updating them for UL and will add the changes here.

aperloff commented on Production/test/dict_cff_Summer20UL16.py:
```diff
@@ -0,0 +1,145 @@
+# format for dict entries:
```
Please remove this file and the other `Production/test/dict_cff_Summer20UL*` files like it. We don't normally add them to the repo.

semrat replied:
I realized we don't have cff_ dictionary in the framework, but I pushed it for checking by you. 
I will remove them.

aperloff commented on WeightProducer/python/MCSamples_Summer20UL16.py:
```diff
@@ -0,0 +1,9 @@
+# 13 TeV miniAOD samples - Summer20UL16
+Summer20UL16samples = [
+    # NB: amcatnlo samples have negative weight events, so NumberEvts = # positive - # negative
+    MCSample('TTToHadronic_TuneCP5_13TeV-powheg-pythia8', '106X_mcRun2_asymptotic_v13-v2', 'RunIISummer20UL16MiniAOD', 'Constant', 112592000, False),
```
Where are all of the other samples? The same question goes for the other years.

semrat replied:
I still not come that point yet, I will update it.
Actually:
- I thought that WeightProducer/python/MCSamples_Summer20UL16.py  file includes the all negative and positive event weight values
- So, I was stuck in the neff issue and did not finish this part yet !

kpedro88 replied:
As we discussed last week, the Neff results are only needed for samples that have negative weights.

aperloff reviewed (commented):
I have reviewed every file. I have made some comments while reviewing the various files. I also have a list of comments here. All of these comments apply to all of the years and APV statuses:

- Do you know what the `MatchEWPDG20` notation in `Summer20UL16/DYJetsToLL_Pt-100To250_MatchEWPDG20_TuneCP5_13TeV-amcatnloFXFX-pythia8_cff.py` means? Same for `WJetsToLNu_Pt-*` samples.
- Is the flat sample (`Summer20UL16/QCD_Pt-15to7000_TuneCP5_Flat2018_13TeV_pythia8_cff.py`) really necessary?
- Why do you need this sample? `Summer20UL16/QCD_Pt-30_MuEnrichedPt4_TuneCP5_13TeV_pythia8_cff.py`
- Do you need both the regular QCD pThat binned and the MuEnriched samples?
- What does `comphep` mean in this sample? `Summer2UL16/ST_t-channel_eleDecays_TuneCP5_13TeV-comphep-pythia8_cff.py` and do we need the samples split up by lepton flavor?
- What do the `CR1` and `CR2` notations mean in:
  - `Summer20UL16/TTTo2L2Nu_TuneCP5CR1_13TeV-powheg-pythia8_cff.py`
  - `Summer20UL16/TTTo2L2Nu_TuneCP5CR2_13TeV-powheg-pythia8_cff.py`
  - Do we need these in addition to the file without that notation?
- Do we need any files with the systematics for the tunes or the "erdON"/"hdampDOWN"/"hdampUP" notations? i.e.
  - `Summer20UL16/TTTo2L2Nu_TuneCP5_erdON_13TeV-powheg-pythia8_cff.py
  - `Summer20UL16/TTTo2L2Nu_TuneCP5down_13TeV-powheg-pythia8_cff.py`
  - `Summer20UL16/TTTo2L2Nu_TuneCP5up_13TeV-powheg-pythia8_cff.py`
  - `Summer20UL16/TTTo2L2Nu_hdampDOWN_TuneCP5_13TeV-powheg-pythia8_cff.py`
  - `Summer20UL16/TTTo2L2Nu_hdampUP_TuneCP5_13TeV-powheg-pythia8_cff.py`
- Where is the `Summer20UL16/WJetsToLNu_HT-2500ToInf` sample?
- Do you need:
  - `Summer20UL16/WZTo3LNu_mllmin01_NNPDF31_TuneCP5_13TeV_powheg_pythia8_cff.py`
  - `Summer20UL16/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8_cff.py`
- Do you have the samples above HT-400 for `Summer20UL16/ZJetsToNuNu_HT-*_TuneCP5_13TeV-madgraphMLM-pythia8_cff.py`? Are there any?
- This is new `Summer20UL16/ZZTo4L_M-1toInf_TuneCP5_13TeV_powheg_pythia8_cff.py`, is it needed?
- Do we need both:
  - `Summer20UL16APV/DYJetsToLL_M-50_TuneCP5_13TeV-amcatnloFXFX-pythia8_cff.py`
  - `Summer20UL16APV/DYJetsToLL_M-50_TuneCP5_13TeV-madgraphMLM-pythia8_cff.py`
- Do you need `Summer20UL16APV/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8_cff.py`?
- Do you need `Summer20UL16APV/tZq_ll_4f_ckm_NLO_TuneCP5CR2_13TeV-amcatnlo-pythia8_cff.py`? It's has a weird systematic tune.
- Where are the rest of the `Summer20UL17/QCD_HT*` samples? I only see `Summer20UL17/QCD_HT1500to2000_TuneCP5_PSWeights_13TeV-madgraph-pythia8_cff.py`.
- Where are the `GJets_DR-0p4_HT-*` samples for Summer20UL16 and Summer20UL16APV?
- Where are the HT binned TTJets samples?

kpedro88 commented:
> Where are the HT binned TTJets samples?

More generally, we need all the Madgraph LO TTJets samples: inclusive, leptonic exclusive (single lepton and dilepton), genMET cut, and HT binned.

semrat commented:
> I have reviewed every file. I have made some comments while reviewing the various files. I also have a list of comments here. All of these comments apply to all of the years and APV statuses:
  Thank you so much for deep review 😃 

>     * Do you know what the `MatchEWPDG20` notation in `Summer20UL16/DYJetsToLL_Pt-100To250_MatchEWPDG20_TuneCP5_13TeV-amcatnloFXFX-pythia8_cff.py` means? Same for `WJetsToLNu_Pt-*` samples.
 I don't know what `MatchEWPDG20`and `WJetsToLNu_Pt-*`,  but I will learn and report back

>     * Is the flat sample (`Summer20UL16/QCD_Pt-15to7000_TuneCP5_Flat2018_13TeV_pythia8_cff.py`) really necessary?
 We don't use this one, we use Pt binned one, I am not sure that is anybody else use it !

>     * Why do you need this sample? `Summer20UL16/QCD_Pt-30_MuEnrichedPt4_TuneCP5_13TeV_pythia8_cff.py`
 We have this for our lagecy analysis, but I don't know where we used that. I will learn and update you.

>     * Do you need both the regular QCD pThat binned and the MuEnriched samples?
 We need at least  QCD pThat one, but as I said we need to confirm MuEnriched by my group.

>     * What does `comphep` mean in this sample? `Summer2UL16/ST_t-channel_eleDecays_TuneCP5_13TeV-comphep-pythia8_cff.py` and do we need the samples split up by lepton flavor?
 I also don't know what is `comphep`, let me learn and update you !

>     * What do the `CR1` and `CR2` notations mean in:
>       
>       * `Summer20UL16/TTTo2L2Nu_TuneCP5CR1_13TeV-powheg-pythia8_cff.py`
>       * `Summer20UL16/TTTo2L2Nu_TuneCP5CR2_13TeV-powheg-pythia8_cff.py`
>       * Do we need these in addition to the file without that notation?
 I have seen some information about CR in CMS-GEN-17-001, but I am not totally sure. Let me get the answer, or maybe @JHiltbrand @cmadrid1 @kpedro88 can answer !

>     * Do we need any files with the systematics for the tunes or the "erdON"/"hdampDOWN"/"hdampUP" notations? i.e.
>       
>       * `Summer20UL16/TTTo2L2Nu_TuneCP5_erdON_13TeV-powheg-pythia8_cff.py
>       * `Summer20UL16/TTTo2L2Nu_TuneCP5down_13TeV-powheg-pythia8_cff.py`
>       * `Summer20UL16/TTTo2L2Nu_TuneCP5up_13TeV-powheg-pythia8_cff.py`
>       * `Summer20UL16/TTTo2L2Nu_hdampDOWN_TuneCP5_13TeV-powheg-pythia8_cff.py`
>       * `Summer20UL16/TTTo2L2Nu_hdampUP_TuneCP5_13TeV-powheg-pythia8_cff.py`
 We need all these notations.

>     * Where is the `Summer20UL16/WJetsToLNu_HT-2500ToInf` sample?
 I am not sure, but I also need to figure out this one like the other missing ones.

>     * Do you need:
>       
>       * `Summer20UL16/WZTo3LNu_mllmin01_NNPDF31_TuneCP5_13TeV_powheg_pythia8_cff.py`
>       * `Summer20UL16/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8_cff.py`
 We will use the "inclusive diboson" samples, but still need to check who else will use the samples.

>     * Do you have the samples above HT-400 for `Summer20UL16/ZJetsToNuNu_HT-*_TuneCP5_13TeV-madgraphMLM-pythia8_cff.py`? Are there any?
I am not sure,  have to check.

>     * This is new `Summer20UL16/ZZTo4L_M-1toInf_TuneCP5_13TeV_powheg_pythia8_cff.py`, is it needed?
 There another separate diboson sample, we are using inclusive one.

>     * Do we need both:
>       
>       * `Summer20UL16APV/DYJetsToLL_M-50_TuneCP5_13TeV-amcatnloFXFX-pythia8_cff.py`
>       * `Summer20UL16APV/DYJetsToLL_M-50_TuneCP5_13TeV-madgraphMLM-pythia8_cff.py`
 We use the madgraph one, if the other one not needed then we can exclude it.

>     * Do you need `Summer20UL16APV/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8_cff.py`?
 We don't use this sample, need to check who else use.

>     * Do you need `Summer20UL16APV/tZq_ll_4f_ckm_NLO_TuneCP5CR2_13TeV-amcatnlo-pythia8_cff.py`? It's has a weird systematic tune.
 Again, we don't use this one.

>     * Where are the rest of the `Summer20UL17/QCD_HT*` samples? I only see `Summer20UL17/QCD_HT1500to2000_TuneCP5_PSWeights_13TeV-madgraph-pythia8_cff.py`.
 Yeah, I recently realized and let me figure where they are 😄 

>     * Where are the `GJets_DR-0p4_HT-*` samples for Summer20UL16 and Summer20UL16APV?
 Another missing ones and will be figured out.

>     * Where are the HT binned TTJets samples?
 They will be existed soon.

semrat commented:
Hi @aperloff, @kpedro88, @cmadrid1  I have checked the lost samples:
**Here is the list of them:**

1. Summer20UL16/WJetsToLNu_HT-2500ToInf 
2. Summer20UL16/ZJetsToNuNu_HT-*_TuneCP5_13TeV-madgraphMLM-pythia8  / **_we need to check for above 400 GeV_**
3. Summer20UL16/GJets_DR-0p4_HT-* 
4. Summer20UL16APV/GJets_DR-0p4_HT-*
5. Summer20UL17/QCD_HT* / **_we need to check for up to HT1500to2000_**
6. Madgraph LO TTJets samples: inclusive, leptonic exclusive (single lepton and dilepton), genMET cut, and HT binned

None of them exists in the DAS.

I will be watching out and will update you.

semrat commented:
Hi @aperloff, @kpedro88, @cmadrid1  I have also checked for this list:

1. Summer20UL16/DYJetsToLL_Pt-100To250_**MatchEWPDG20**_TuneCP5_13TeV-amcatnloFXFX-pythia8
2. Summer20UL16/ST_t-channel_eleDecays_TuneCP5_13TeV-**comphep**-pythia8:
3. Summer20UL16/TTTo2L2Nu_**TuneCP5CR1**_13TeV-powheg-pythia8
4. Summer20UL16/TTTo2L2Nu_**TuneCP5CR2**_13TeV-powheg-pythia8
5. Summer20UL16APV/tZq_ll_4f_ckm_NLO_TuneCP5CR2_13TeV-amcatnlo-pythia8 

**MatchEWPDG20**
- I could not find anything 😞 

**comphep:**
- it is an alternative generator
- if we have madgraph or powheg samples, it will be more sufficient and we don’t need comphep
- I also found the source code if it will be useful: https://cmssdt.cern.ch/lxr/source/GeneratorInterface/GenFilters/plugins/ComphepSingletopFilterPy8.cc


**for the CR1 and CR2**
- here is the information about them: https://twiki.cern.ch/twiki/bin/viewauth/CMS/TopSystematics#Parton_shower_Color_Reconnection
- QCD-inspired (CR1)
- Gluon move (CR2)
- I am not sure we need them or not !!!

Please let me know, will I exclude these samples or not ?
Thank you for your helps.

semrat commented:
Hi @aperloff, @cmadrid1, @kpedro88 I have the last one 😄 
**Here is the list of the samples which  Alexx asked me if we will include them or not:**

1. Summer20UL16/QCD_Pt-15to7000_TuneCP5_Flat2018_13TeV_pythia8
2. Summer20UL16/WZTo3LNu_mllmin01_NNPDF31_TuneCP5_13TeV_powheg_pythia8
3. Summer20UL16/ZZTo4L_M-1toInf_TuneCP5_13TeV_powheg_pythia8 / **_separate diboson, we use inclusive one_**
4. Summer20UL16/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8 / _**separate diboson, we use inclusive one**_
5. Summer20UL16APV/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8 / **_separate diboson, we use inclusive one_**
6. Summer20UL16APV/DYJetsToLL_M-50_TuneCP5_13TeV-amcatnloFXFX-pythia8 / **_we use the mad graph one_**

Kevin, do you know who will use these samples, we are not using them for our analyis
Or is there anybody who I need to check with him/her ?

aperloff commented:
> **comphep:**
>
>     * it is an alternative generator
>
>     * if we have madgraph or powheg samples, it will be more sufficient and we don’t need comphep
> 
>     * I also found the source code if it will be useful: https://cmssdt.cern.ch/lxr/source/GeneratorInterface/GenFilters/plugins/ComphepSingletopFilterPy8.cc
> 
> 
> **for the CR1 and CR2**
> 
>     * here is the information about them: https://twiki.cern.ch/twiki/bin/viewauth/CMS/TopSystematics#Parton_shower_Color_Reconnection
> 
>     * QCD-inspired (CR1)
> 
>     * Gluon move (CR2)
> 
>     * I am not sure we need them or not !!!

If you don't need them and there are alternative "standard" samples, then please include the standard ones and exclude these new samples.

aperloff commented:
> **MatchEWPDG20**
> 
>     * I could not find anything 😞

I'm told that this is an update of the EWK values. A little more information can be found at https://github.com/cms-sw/genproductions/blob/master/bin/GenValidation/validation_cards/dylljetsmlm/templates/customizecards.dat.

aperloff commented:
> **Here is the list of the samples which Alexx asked me if we will include them or not:**
> 
>     1. Summer20UL16/QCD_Pt-15to7000_TuneCP5_Flat2018_13TeV_pythia8
> 
>     2. Summer20UL16/WZTo3LNu_mllmin01_NNPDF31_TuneCP5_13TeV_powheg_pythia8
> 
>     3. Summer20UL16/ZZTo4L_M-1toInf_TuneCP5_13TeV_powheg_pythia8 / **_separate diboson, we use inclusive one_**
> 
>     4. Summer20UL16/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8 / _**separate diboson, we use inclusive one**_
> 
>     5. Summer20UL16APV/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8 / **_separate diboson, we use inclusive one_**
> 
>     6. Summer20UL16APV/DYJetsToLL_M-50_TuneCP5_13TeV-amcatnloFXFX-pythia8 / **_we use the mad graph one_**

The only one I think we might want is 6. Please remove the others. If we need to add them later we can do so, but I'd like to avoid prematurely adding samples we don't need.

semrat commented:
> > **Here is the list of the samples which Alexx asked me if we will include them or not:**
> > ```
> > 1. Summer20UL16/QCD_Pt-15to7000_TuneCP5_Flat2018_13TeV_pythia8
> > 
> > 2. Summer20UL16/WZTo3LNu_mllmin01_NNPDF31_TuneCP5_13TeV_powheg_pythia8
> > 
> > 3. Summer20UL16/ZZTo4L_M-1toInf_TuneCP5_13TeV_powheg_pythia8 / **_separate diboson, we use inclusive one_**
> > 
> > 4. Summer20UL16/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8 / _**separate diboson, we use inclusive one**_
> > 
> > 5. Summer20UL16APV/ZGToLLG_01J_5f_TuneCP5_13TeV-amcatnloFXFX-pythia8 / **_separate diboson, we use inclusive one_**
> > 
> > 6. Summer20UL16APV/DYJetsToLL_M-50_TuneCP5_13TeV-amcatnloFXFX-pythia8 / **_we use the mad graph one_**
> > ```
> 
> The only one I think we might want is 6. Please remove the others. If we need to add them later we can do so, but I'd like to avoid prematurely adding samples we don't need.

Sounds good.

semrat commented:
> > **comphep:**
> > ```
> > * it is an alternative generator
> > 
> > * if we have madgraph or powheg samples, it will be more sufficient and we don’t need comphep
> > 
> > * I also found the source code if it will be useful: https://cmssdt.cern.ch/lxr/source/GeneratorInterface/GenFilters/plugins/ComphepSingletopFilterPy8.cc
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > **for the CR1 and CR2**
> > ```
> > * here is the information about them: https://twiki.cern.ch/twiki/bin/viewauth/CMS/TopSystematics#Parton_shower_Color_Reconnection
> > 
> > * QCD-inspired (CR1)
> > 
> > * Gluon move (CR2)
> > 
> > * I am not sure we need them or not !!!
> > ```
> 
> If you don't need them and there are alternative "standard" samples, then please include the standard ones and exclude these new samples.

Then I am excluding them

semrat commented:
> > **MatchEWPDG20**
> > ```
> > * I could not find anything 😞
> > ```
> 
> I'm told that this is an update of the EWK values. A little more information can be found at https://github.com/cms-sw/genproductions/blob/master/bin/GenValidation/validation_cards/dylljetsmlm/templates/customizecards.dat.

I realized that it seems like that **_DYJetsToLL_Pt-*_** and **_WJetsToLNu_Pt-*_**  only the version of the Pt-binned. So, we need to include them

kpedro88 commented:
I'm not aware of any analyses that use pt-binned WJets or DYJets; the HT-binned samples are more commonly used. Is there anyone actually asking for those samples?

semrat commented:
> I'm not aware of any analyses that use pt-binned WJets or DYJets; the HT-binned samples are more commonly used. Is there anyone actually asking for those samples?

We are using the HT one, but I am not who else we need to check with !

kpedro88 commented:
In general, if we're not using certain samples in pre-legacy and we're not aware of specific requests for them in ultra-legacy, then I don't think we should include them.

semrat commented:
> > > **comphep:**
> > > ```
> > > * it is an alternative generator
> > > 
> > > * if we have madgraph or powheg samples, it will be more sufficient and we don’t need comphep
> > > 
> > > * I also found the source code if it will be useful: https://cmssdt.cern.ch/lxr/source/GeneratorInterface/GenFilters/plugins/ComphepSingletopFilterPy8.cc
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > **for the CR1 and CR2**
> > > ```
> > > * here is the information about them: https://twiki.cern.ch/twiki/bin/viewauth/CMS/TopSystematics#Parton_shower_Color_Reconnection
> > > 
> > > * QCD-inspired (CR1)
> > > 
> > > * Gluon move (CR2)
> > > 
> > > * I am not sure we need them or not !!!
> > > ```
> > 
> > 
> > If you don't need them and there are alternative "standard" samples, then please include the standard ones and exclude these new samples.
> 
> Then I am excluding comphep ones, but let me confirm by my group if we need to CR1 and CR2 !

semrat commented:
> In general, if we're not using certain samples in pre-legacy and we're not aware of specific requests for them in ultra-legacy, then I don't think we should include them.

Then MatchEWPDG20 will be also excluded

aperloff commented:
> Then MatchEWPDG20 will be also excluded

You may want to include those over the non-`MatchWEPDG20` samples, if we have them for all three years. I was recently pointed to https://indico.cern.ch/event/981801/contributions/4176522/attachments/2172357/3667733/210115_shKo_GEN.pdf, which makes it seem like the new samples are an improvement, though not perfect. @kpedro88 how do you feel about including these newly tuned samples? I hate to introduce duplicate samples or an x-factor in terms of potential generator problems, but this one might be worth it.

kpedro88 commented:
@aperloff as I said above, I'm not aware of any TreeMaker-based analyses that use WJets or DYJets pt-binned samples. If nobody explicitly needs and plans to use these samples, I'd rather we not do all the work associated with adding them for no reason.

binayab committed:
Updated the scenarios for UL 2016,17 and 18

binayab committed:
Downloaded the UL update corrections to jec, jer, pileup, JSON , to be used in scenarios.py

kpedro88 commented:
@binayab some comments on these recent updates:
1. Can you post here exactly which commands you used to generate the pileup reweighting histograms? (Both so we can double-check them, and in case we need to update the histograms in the future.)
2. We had discussed checking the combined .db file `Summer19UL18_V5_DATA.db` for the 2018 data JECs, to see if it gives the same results as the separate run period files. Did this not work?
3. The list of new lepton-based triggers seems very long. For what will these be used? Are they really all needed?
4. Some of the JER .db files did not download correctly; please recheck them. (Anything in https://github.com/TreeMaker/TreeMaker/commit/212d0e80f7e5ea55729ea506470f7e000c81a065 that is a .db file but shows up on that webpage without "BIN" next to the title needs to be fixed. For example, https://github.com/TreeMaker/TreeMaker/blob/212d0e80f7e5ea55729ea506470f7e000c81a065/Production/test/data/jer/Summer19UL17_JRV2_DATA.db)

binayab commented:
Hi @kpedro88 

for 4) I will look into it and fix that issue. **Fixed that now.**
2) I have not been able to do the checking yet, waiting on last piece from Semra before I can use unitTest.py. So, the one that I have there is a placeholder until I finish that task. 
3) I updated the triggerlist removing the duplicates and sorting alphabetically. There are additional 52 dilepton triggerlist. We are not sure at this moment if we need all of them but we will need all of them for a trigger study in future. Will push that soon.  **Pushed the latest and greatest final list.** 
1) So, for the updated pileup histograms, I downloaded the histograms as mentioned in the twiki page instead of generating using pileupCalc.py. https://twiki.cern.ch/twiki/bin/view/CMS/PileupJSONFileforData 
If we need to produce the ROOT histograms ourselves, please let us know. In the meantime, I will be looking into those histograms. Thank you.

binayab committed:
Updated the Triggerlist by removing the duplicates and sorting alphabetically

semrat commented:
Hi @aperloff, @kpedro88 
I have been checking the samples which were not exist yet in the DAS.

I realized that there is no **_GJets_DR-0p4_HT-*_**: 

- instead of this there is **_GJets_HT-*_**
- How I can be sure that they are different samples !


For the TTJets:

- There is no TTJets - **_madgraph_** one in the DAS. 
- We only have TTJets - **_amcatnlo-pythia8_**

Rest of the **_lost samples_** are exist anymore.

Thank you both for your helps.

binayab committed:
final triggerlist

binayab committed:
replaced the wrong JER files with correct ones

binayab committed:
fully updated triggerlist didn't get pushed when i did earlier. This is the final one

JHiltbrand commented:
Hello @kpedro88 and @aperloff,

In working with the sample lists that we have been generating, I realized we are picking out the `v1` versions of the 20UL MiniAOD. I see that there is ongoing generation of a `v2` MiniAOD, presumably it is preferred to use that version for all samples and all years ? I guess based on [*], the changes targeted for `v2` are [**], in case it is unclear what the differences are.

[*] https://hypernews.cern.ch/HyperNews/CMS/get/physics-validation/3658.html
[**] https://github.com/cms-sw/cmssw/issues/27889

aperloff commented:
I would think we would want the MiniAODv2 samples, but it sounds like the re-MiniAOD isn't done yet for Summer20UL*17 or 18 (example: https://cms-pdmv.cern.ch/grasp/existing?name=RunIISummer20UL17*GEN&pwg=SUS&dataset=QCD_HT). The `-v1` and `-v2` notations after the GT are computing notations for when they have to restart the processing of a given dataset. We would indeed prefer the "version" with the highest number, usually.

JHiltbrand commented:
Hi @aperloff,

Thanks for the input ! We will start collecting things for the `MiniAODv2` going forward.

On a related note, is there any support in the TreeMaker framework to run `xsecfinder_cfg.py` in batch for all these UL samples for the xs info ? I can imagine that most of these numbers should stay the same, so hopefully we do not have to rerun too much.

aperloff commented:
> is there any support in the TreeMaker framework to run `xsecfinder_cfg.py` in batch for all these UL samples for the xs info ? I can imagine that most of these numbers should stay the same, so hopefully we do not have to rerun too much.

No, not really. Though I will say that these run fairly fast interactively, for most samples. Don't assume the XS will remain the same. We have been surprised by large changes in the past. It's always good to check.

JHiltbrand commented:
Hi @aperloff and @kpedro88,

I was reviewing lepton IDs with @binayab, and we noted that the definition of medium ID for muons currently in the framework includes selections on dB and dZ  [1], which I guess are found in the tight ID WP as well. We could not find this suggestion on various muon POG pages, though the variable name `TreeMaker` suggests it came out of a SUSY recommendation. We were wondering if you knew if these additional requirements should carry into UL ?

For electron IDs, we find that the recommended cut based IDs [2] can be accessed from the `pat::Electron` object---the same ID that appears to be manually implemented in `LeptonProducer.cc` [3]. If our assessment about the ID is correct, would it prove useful to simplify [3] to just a few calls to the `pat::Electron`s `electronID` method ?

[1] https://github.com/TreeMaker/TreeMaker/blob/Run2_UL/Utils/src/LeptonProducer.cc#L393
[2] https://twiki.cern.ch/twiki/bin/view/CMS/CutBasedElectronIdentificationRun2#Cut_Based_Electron_ID_for_Run_2
[3] https://github.com/TreeMaker/TreeMaker/blob/Run2_UL/Utils/src/LeptonProducer.cc#L413-L485

binayab committed:
updated goodjetsproducer with new set of values as recommended for UL

aperloff commented:
@JHiltbrand We implement the lepton IDs manually, using the cut based scheme so that we can update more quickly when an ID changes. If using the `pat::Electron`'s built in IDs we would need to wait for an update to CMSSW and likely a miniAOD reprocessing, would could take a very long time. Additionally, some of the muon ID code has been poorly written in the past. It may have been improved recently, but we would prefer not to add buggy/slow external tools.

To your other question about the dB and dZ cuts, they come from [4]. You can see that there are other analyses using them in [5]. This twiki makes no mention of UL, so it might be worth bringing up with the SUSY conveners [6] or the SUSY muon contact [7].

[4] https://twiki.cern.ch/twiki/bin/viewauth/CMS/SUSLeptonSF#ID_IP_ISO
[5] https://twiki.cern.ch/twiki/bin/viewauth/CMS/SUSLeptonSF#Full_Run_2_analyses
[6] https://twiki.cern.ch/twiki/bin/view/CMS/SUSY
[7] https://twiki.cern.ch/twiki/bin/view/CMS/SusyObjectExperts

JHiltbrand commented:
Hi Alexx,

Sounds good about the implementation of lepton IDs and we will get in contact with [6] and [7] to clarify what should be done for UL.

Commit:
add TMEra for 2016APV

Commit:
update prefiring module to include muon prefire

Commit:
fix syntax for jet producer cfi

JHiltbrand commented:
Hello @aperloff, @kpedro88,

In e552f2b, I update the prefiring module to include muon prefiring in the calculations of prefiring. The changes follow how things are done in CMSSW. Please let me know if what I did is reasonable.

Commit:
cleanup printout for prefiring module

Commit:
protect prefire module when no calculation needed

kpedro88 commented on TreeMaker/python/makeTreeFromMiniAOD_cff.py:
```diff
@@ -979,22 +979,41 @@ def makeTreeFromMiniAOD(self,process):
+    TMeras.TMUL2017.toModify(prefiringDataEraMuon,    value = "20172018")
+    TMeras.TMUL2018.toModify(prefiringDataEraMuon,    value = "20172018")
+
+    if len(prefiringDataEraECAL.value.value())>0 and len(prefiringDataEraMuon.value.value())>0:
```
rather than extra PSets and if statements, just create `process.L1NonPrefiringProbProducer` with default values and then apply `TMeras.TMUL2018.toModify(process.L1NonPrefiringProbProducer, ...)` directly.

JHiltbrand replied:
Yeah, will simplify.

kpedro88 commented on Utils/src/L1NonPrefiringProbProducer.cc:
```diff
@@ -0,0 +1,348 @@
+  TFile* file_ecalprefiringmaps = TFile::Open(fnameecal.c_str(),"read");
+
+  std::string mapphotonfullname = "L1prefiring_photonptvseta_"+ dataeraecal_; 
+  file_ecalprefiringmaps->ls();
```
remove printout

JHiltbrand replied:
Cleaned up in next commit.

kpedro88 commented on Utils/src/L1NonPrefiringProbProducer.cc:
```diff
@@ -0,0 +1,348 @@
+
+   double abseta = abs(eta);
+   double prefrate = 0.0; double preferr = 0.0;
+   if ((dataeramuon_.find("2016") != std::string::npos) && (eta > 1.24 && eta < 1.6) && (phi > 2.44346 && phi < 2.79253)){
```
string comparisons are expensive. since `dataeramuon_` is set in the constructor and does not change, just do this comparison in the constructor and keep a bool indicating that the hotspot function should be used.

JHiltbrand replied:
Done, in next commit

kpedro88 commented on Utils/src/L1NonPrefiringProbProducer.cc:
```diff
@@ -0,0 +1,348 @@
+   } else if (abseta < 2.25) {
+      prefrate = h_prefparam_muon_2p10to2p25->Eval(pt);
+      preferr = h_prefparam_muon_2p10to2p25->GetParError(2);
+   } else if (abseta < 2.4) {
```
this if/else if structure is essentially a linear search (O(N)) over bins. instead, the search can be done in O(logN) by making an `std::map<double,TF1*>`, where the keys are the bin upper boundaries. Then use the `std::map` accessor `upper_bound()` to get the right bin.

JHiltbrand replied:
Addressed in e9088f6

kpedro88 commented on Utils/src/L1NonPrefiringProbProducer.cc:
```diff
@@ -0,0 +1,348 @@
+  std::string mapmuonfullname = "L1prefiring_muonparam_HotSpot_" + dataeramuon_;
+  h_prefparam_muon_hotspot = (TF1*)file_muonprefiringparams->Get(mapmuonfullname.c_str());
+
+  mapmuonfullname = "L1prefiring_muonparam_0.0To0.2_" + dataeramuon_;
```
to reduce duplication, make `std::vector<std::pair<double,std::string>>` and then loop over it to fill the map suggested in the previous comment.

JHiltbrand replied:
Compactified in e9088f6

kpedro88 commented on Utils/src/L1NonPrefiringProbProducer.cc:
```diff
@@ -0,0 +1,348 @@
+  useEMpt_(iConfig.getParameter<bool>("UseJetEMPt")),
+  prefiringRateSystUncECAL_(iConfig.getParameter<double>("PrefiringRateSystematicUnctyECAL")),
+  prefiringRateSystUncMuon_(iConfig.getParameter<double>("PrefiringRateSystematicUnctyMuon")),
+  jet_max_muon_fraction_(iConfig.getParameter<double>("JetMaxMuonFraction"))
```
this is not used. should it be used somewhere? if not, delete it.

JHiltbrand replied:
Thanks for the check, it should be used, but I forgot to implement, will appear in next commit.

kpedro88 commented on Utils/src/L1NonPrefiringProbProducer.cc:
```diff
@@ -0,0 +1,348 @@
+#include <array>
+#include <cmath>
+
+//based on: https://github.com/lathomas/cmssw/blob/L1Prefiring_9_4_9/L1Prefiring/EventWeightProducer/plugins/L1NonPrefiringProbProducer.cc
```
update this comment to point to the latest official version w/ muon prefiring

JHiltbrand replied:
Done, will appear in next commit

aperloff commented:
> one more comment: the use of TF1 objects means this should be changed to a stream module, since TF1 is not fully thread safe.

Would it be worth spending some time in the constructor to convert from TF1 to `reco::FormulaEvaluator` (https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/CommonTools/Utils/interface/FormulaEvaluator.h)? Since the TF1 would only be in the constructor, would that get around the need to use a stream over a global producer?

kpedro88 commented:
> Would it be worth spending some time in the constructor to convert from TF1 to reco::FormulaEvaluator (https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/CommonTools/Utils/interface/FormulaEvaluator.h)? Since the TF1 would only be in the constructor, would that get around the need to use a stream over a global producer?

Yes, then a global producer could be used. As you know, this requires careful validation of all functional forms used to ensure that ROOT and FormulaEvaluator give consistent output. Given that, I probably would not prioritize it, but if someone wants to take it on, it would be nice to have (and maybe to contribute upstream to CMSSW).

Commit:
address comments from Kevin and Alexx on prefire module

JHiltbrand commented:
> 
> 
> > Would it be worth spending some time in the constructor to convert from TF1 to reco::FormulaEvaluator (https://github.com/cms-sw/cmssw/blob/6d2f66057131baacc2fcbdd203588c41c885b42c/CommonTools/Utils/interface/FormulaEvaluator.h)? Since the TF1 would only be in the constructor, would that get around the need to use a stream over a global producer?
> 
> Yes, then a global producer could be used. As you know, this requires careful validation of all functional forms used to ensure that ROOT and FormulaEvaluator give consistent output. Given that, I probably would not prioritize it, but if someone wants to take it on, it would be nice to have (and maybe to contribute upstream to CMSSW).

For the moment, I just converted to an edm::stream::EDProducer along with addressing the other comments in e9088f6.

kpedro88 commented:
One followup comment on https://github.com/TreeMaker/TreeMaker/pull/573#issuecomment-902297435: we prefer whichever sample has the status VALID, which is usually the one with the highest v#, but not always.

Commit:
point jet producer config to UL eras

Commit:
remove old pre-UL eras

Commit:
remove ref to TM80, remove CSVv2, update DeepCSV WPs

Commit:
remove reference to old MC samples

Commit:
remove reference to TM80X era

Commit:
simplify now that TM80 is gone

Commit:
remove all pre-UL scenarios, update PU weights histos with proper format

Commit:
remove reference to TM80X, default b tag is DeepCSV, remove all old unit tests

Commit:
no special recipe in UL to rerun certain ECAL MET filters, added muon dZ

kpedro88 commented on Production/test/unitTest.py:
```diff
@@ -93,35 +93,10 @@ def defineTests(mytests, scenario, name, numevents, command, dataset, inputFiles
     mytests.append(Test(scenario,name,numevents,command,dataset,inputFilesConfig,nstart=0,nfiles=10,redir=""))
 
     # pre-defined tests
-    mytests.append(Test("Summer16v3","Summer16v3.GJets_HT-600ToInf_TuneCUETP8M1_13TeV-madgraphMLM-pythia8",numevents,command,inputFilesConfig="Summer16v3.GJets_HT-600ToInf_TuneCUETP8M1_13TeV-madgraphMLM-pythia8",nstart=0,nfiles=1))
```
we have unit tests for GJets as well as ttbar in order to exercise the cleaned jet reclustering more thoroughly, so please add some for UL at some point (in addition to the signal and data ones, whenever those samples are available)

kpedro88 commented on TreeMaker/python/makeTreeFromMiniAOD_cff.py:
```diff
@@ -862,28 +797,14 @@ def makeTreeFromMiniAOD(self,process):
-        softDropMass = cms.vstring('SoftDrop'),
-        subjets = cms.vstring('SoftDrop'),
-        SJbDiscriminatorCSV = cms.vstring('SoftDrop', 'pfCombinedInclusiveSecondaryVertexV2BJetTags'),
+        doECFs = True, # temporarily disabled
```
this line can just be removed

binayab committed:
Included the jec for the Summer19UL18 DATA combined db

Commit:
dictionaries not in separate folder, in main

Commit:
remove old file and sample dictionaries

Commit:
remove old data file lists

Commit:
remove old pileup stuff

Commit:
remove old file lists

Commit:
update UL samples, xsection, lists, dictionaries, etc for latest 20UL MC

Commit:
update MET filters in config

Commit:
remove old jec and jer db files

Commit:
minor fixes to scenarios, add some unit tests

Commit:
address deprecation warning about CondDB

JHiltbrand commented:
Hi @kpedro88 and @aperloff,
When running unit tests on the changes in this PR, I observe the following issue with applying JECs, concerning corrections for a specifically named jet collection. Is there anything obvious we could have missed to yield this message.

```
%MSG
Begin processing the 1st record. Run 1, Event 3258011, LumiSection 3259 on stream 0 at 13-Sep-2021 07:52:11.196 CDT
%MSG-w L3Absolute not found:   PATJetUpdater:updatedPatJetsSlimmedJetsAK8PFPuppiSoftDropPackedSubJetsUpdatedJEC  13-Sep-2021 07:52:20 CDT Run: 1 Event: 3258005
L2L3Residual and L3Absolute are not part of the jetCorrFactors
of module patJetCorrFactorsSlimmedJetsAK8PFPuppiSoftDropPackedSubJetsUpdatedJEC. Jets will remain uncorrected.
%MSG
%MSG-w L3Absolute not found:  PATJetUpdater:updatedPatJetsAK8UpdatedJEC  13-Sep-2021 07:52:20 CDT Run: 1 Event: 3258005
L2L3Residual and L3Absolute are not part of the jetCorrFactors
of module patJetCorrFactorsAK8UpdatedJEC. Jets will remain uncorrected.
%MSG
13-Sep-2021 07:52:24 CDT  Closed file root://cmsxrootd.fnal.gov//store/mc/RunIISummer20UL16MiniAODAPVv2/TTToHadronic_TuneCP5_13TeV-powheg-pythia8/MINIAODSIM/106X_mcRun2_asymptotic_preVFP_v11-v1/120000/0343A84D-6182-D149-B7E1-2B824263CC61.root
```

aperloff commented:
@JHiltbrand Those warnings are normal. You can ignore them.

kpedro88 commented on Utils/python/pdfweightproducer_cfi.py:
```diff
@@ -8,6 +8,6 @@
 
 from TreeMaker.TreeMaker.TMEras import TMeras
-(TMeras.TM2017 | TMeras.TM2018).toModify(PDFWeightProducer,
+(TMeras.TMUL2017 | TMeras.TMUL2018).toModify(PDFWeightProducer,
```
this modification probably isn't needed anymore - it specifically handled the change in PDFs for pre-legacy MC in 2017. We should check the number of replicas for the UL PDFs and make sure the right number is used for all years.

JHiltbrand replied:
Yeah, I dumped the headers for a sample across the years and now see that things are the same. So I updated in d01ea7d to simplify things down.

Commit:
all years same for pdfs

JHiltbrand commented:
Dear @kpedro88 and @aperloff,

I believe at this time, we have addressed most things cited in #571 (that we obviously needed/agreed to update) and I have an updated reply in that thread detailing the status in [1]. For the moment, I have no major additional changes to push, so I think now is a good time to review the PR as a whole to determine next steps. From there we can iterate on additional items, more extensive testing, etc, based on what you guys think. Thank you for the continued feedback on our efforts.

[1] https://github.com/TreeMaker/TreeMaker/issues/571#issuecomment-909789127

kpedro88 commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -1805,7 +1660,9 @@ def get_value(self, name, energy, year, key, strict=False):
         fields = [self.__key_field_map[key][0]+"_"+energy,self.__key_field_map[key][0]+"_"+year]
         if not name in self.__values_dict:
-            raise KeyError("ERROR MCSampleValuesHelper::Unknown process \"" + str(name) + "\"")
+            print("ERROR MCSampleValuesHelper::Unknown process \"" + str(name) + "\"")
```
why does this no longer raise an exception?

JHiltbrand replied:
Ah, that is an oversight on my part when I pushed my changes onto the branch. I will restore the exception raising.

JHiltbrand replied:
Reverted in 4d2e518

kpedro88 commented on WeightProducer/python/MCSamples_Summer20UL16.py:
```diff
@@ -0,0 +1,105 @@
+# 13 TeV miniAOD samples - Summer20UL16
+Summer20UL16samples = [
+    # NB: amcatnlo samples have negative weight events, so NumberEvts = # positive - # negative
+    MCSample('WGJets_MonoPhoton_PtG-40to130_TuneCP5_13TeV-madgraph-pythia8', '106X_mcRun2_asymptotic_v17-v2', 'RunIISummer20UL16MiniAODv2', 'Constant', 1386165, False),
```
one of the remaining todo items is to remove the `WrongPU` field from `MCSample` (that's the `False` at the end of this line)

JHiltbrand replied:
When I remove this, I'll also remove the 2017 METfix as it is no longer needed in UL per [1].

[1] https://twiki.cern.ch/twiki/bin/view/CMS/MissingETUncertaintyPrescription#Instructions_for_2017_data_with

kpedro88 replied:
Yes, very good.

JHiltbrand replied:
Now addressed in ea9d776

kpedro88 commented on WeightProducer/python/MCSamples_Summer20UL16.py:
```diff
@@ -0,0 +1,105 @@
+Summer20UL16samples = [
+    # NB: amcatnlo samples have negative weight events, so NumberEvts = # positive - # negative
+    MCSample('WGJets_MonoPhoton_PtG-40to130_TuneCP5_13TeV-madgraph-pythia8', '106X_mcRun2_asymptotic_v17-v2', 'RunIISummer20UL16MiniAODv2', 'Constant', 1386165, False),
+    MCSample('WWG_TuneCP5_13TeV-amcatnlo-pythia8', '106X_mcRun2_asymptotic_v17-v1', 'RunIISummer20UL16MiniAODv2', 'Constant', 698000, False),
```
amcatnlo samples should have the `NumberEvtsDiff` field filled because they have negative weights

JHiltbrand replied:
Good catch, I'll fix that one and see if I missed any others.

JHiltbrand replied:
Corrected in ea9d776

Commit:
these should not go in repo

Commit:
remove use of wrongPU, METfix; correct MCSamples for neff

Commit:
raise exception as before rather than just print

JHiltbrand commented:
Dear @aperloff and @kpedro88,

@cmadrid1 and I discussed the current changes proposed in this PR, to address #571, and we believe it is in a good state to be merged. Last week I addressed some comments from Kevin from initial review. We comment on the status of the various changes in [1], where only remaining items are about adding additional object collections, which I guess needs further input from the groups that will use them. Please let us know if there are things to be done and what you would like to see for the next steps.

[1] https://github.com/TreeMaker/TreeMaker/issues/571#issuecomment-909789127

aperloff commented on Production/python/scenarios.py:
```diff
@@ -5,319 +5,80 @@ class Scenario:
+        if sname == "Summer20UL16":
             self.set_vars(
-                globaltag="94X_mcRun2_asymptotic_v3",
+                globaltag="106X_mcRun2_asymptotic_v17", #changed v15 to v17
```
Remove the comment. It is unnecessary.

aperloff commented on Production/python/scenarios.py:
```diff
@@ -5,319 +5,80 @@ class Scenario:
-                jecfile=self.absdatapath+"/jec/Summer16_07Aug2017_V11_MC",
-                jerfile=self.absdatapath+"/jer/Summer16_25nsV1_MC",
-                pufile=self.reldatapath+"/PileupHistograms_2016_69mb_pm5.root",
+                jecfile=self.absdatapath+"/jec/Summer19UL16_V7_MC",
```
Since the Summer19UL JEC are the only ones available, it is fine to have them here. However, you should make a separate issue about updating these to Summer20UL when they become available. I'm afraid the issue will be lost in #571.

JHiltbrand replied:
Indeed, a new clean issue #584 will keep track of these remaining items.

aperloff commented on Production/python/scenarios.py:
```diff
@@ -5,319 +5,80 @@ class Scenario:
-                era="Run2_2018",
                 localera="TMUL2018",
             )
         else: # if no recognized scenario, cannot go forward
```
What about scenarios for:
- signal samples (missing completely)
- FastSim (missing completely)
- data (missing for UL2016)
Don't you anticipate needing these?

JHiltbrand replied:
Added these additional scenarios. No JEC/JERs for FastSim it seems at the moment. Will denote in #584.

aperloff commented on Production/python/scenarios.py:
```diff
@@ -5,319 +5,80 @@ class Scenario:
+                localera="TMUL2018",
             )
-        elif sname == "2018ReReco17Sep":
+        elif sname == "Summer20UL18_DATA":  #naming convention for DATA "_DATA"
```
Remove the comment about the naming convention for the data scenarios and the ones that say `#no change`.

aperloff commented on Production/test/condorSub/dict_Summer20UL16APV_diboson.py:
```diff
@@ -0,0 +1,24 @@
+flist = {
```
Anyone else besides me find it weird that we name this dictionary "diboson" when it clearly has "triboson" samples?

kpedro88 replied:
Personally I'm okay with keeping these in the same dictionary.

aperloff commented on Production/test/condorSub/dict_Summer20UL16APV_dyjets.py:
```diff
@@ -0,0 +1,11 @@
+
+       ['Summer20UL16APV.DYJetsToLL_M-50_TuneCP5_13TeV-amcatnloFXFX-pythia8'],
+       ['Summer20UL16APV.DYJetsToLL_M-50_TuneCP5_13TeV-madgraphMLM-pythia8'],
+       ['Summer20UL16APV.DYJetsToLL_M-50_TuneCP5_13TeV-madgraphMLM-pythia8'],
```
Isn't this line a duplicate of the one above it?

aperloff commented on Production/test/condorSub/dict_Summer20UL16APV_rpv.py:
```diff
@@ -0,0 +1,8 @@
+flist = {
+
+   "scenario": "Summer20UL16APV",
```
Won't these be using a signal scenario?

aperloff commented on Production/test/condorSub/dict_Summer20UL16APV_shh.py:
```diff
@@ -0,0 +1,8 @@
+flist = {
+
+   "scenario": "Summer20UL16APV",
```
Same comment about the signal scenarios. I'm going to stop repeating myself here.

aperloff commented on Production/test/condorSub/dict_Summer20UL16APV_tth.py:
```diff
@@ -0,0 +1,19 @@
+flist = {
```
This are probably better put into the `_tt.py` file and then you can remove the `_tth.py` file. Then we will only have the `_tt.py` and `_ttbar.py` files and the grouping will make more sense.

aperloff commented on Production/test/condorSub/dict_Summer20UL16APV_ttjets.py:
```diff
@@ -0,0 +1,8 @@
+flist = {
```
I don't think we need both `_ttbar.py` and `_ttjets.py` files. Seems redundant to me.

aperloff commented on Production/test/unitTest.py:
```diff
@@ -93,35 +93,12 @@ def defineTests(mytests, scenario, name, numevents, command, dataset, inputFiles
     mytests.append(Test(scenario,name,numevents,command,dataset,inputFilesConfig,nstart=0,nfiles=10,redir=""))
 
     # pre-defined tests
```
What about unit tests for signal, data, and FastSim?

JHiltbrand replied:
Can be done, added to the list in #584.

aperloff commented on TreeMaker/python/makeJetVars.py:
```diff
@@ -108,29 +102,24 @@ def makeJetVars(self, process, JetTag, suff, storeProperties, SkipTag=cms.VInput
         JetTag       = HTJetsTag,
         BTagInputTag = cms.string('pfDeepCSVDiscriminatorsJetTags:BvsAll'),
-        BTagCutValue = cms.double(0.6321)
+        BTagCutValue = cms.double(0.6321) #TODO: wait for UL16 recommendations
```
Don't write `TODO`s in the code. Instead, make a GitHub issue.

kpedro88 replied:
I don't mind having clear TODOs in code, but in this case a GitHub issue is also a good idea.

JHiltbrand replied:
New issue #584 to keep track of latest list of TODOs.

aperloff commented on TreeMaker/python/makeTreeFromMiniAOD_cff.py:
```diff
@@ -243,19 +242,17 @@ def makeTreeFromMiniAOD(self,process):
         #get name of JECs without any directories
         JECera = self.jecfile.split('/')[-1]
-        JECPatch = cms.string('sqlite_file:'+self.jecfile+'.db')
+        process.CondDB.connect = cms.string('sqlite_file:'+self.jecfile+'.db')
```
Have you tested that this pattern, setting the `connect` variable outside the ESSource, actually works?

kpedro88 replied:
I dumped the config after this change to check this, and it seems to work correctly because each actual `ESSource` is created by *cloning* `process.CondDB`.

aperloff commented on TreeMaker/python/makeTreeFromMiniAOD_cff.py:
```diff
@@ -862,28 +801,13 @@ def makeTreeFromMiniAOD(self,process):
         JetTag=JetAK8Tag,
         suff='AK8',
         storeProperties=2,
-        doECFs = not TMeras.TM80X._isChosen(), # temporarily disabled
```
Does this mean that `doECFs` is set to true by default file and thus true all of the time?

kpedro88 replied:
Right now, yes. I think it's still useful to have the option to disable it for testing.

aperloff commented on Utils/python/goodjetsproducer_cfi.py:
```diff
@@ -32,44 +34,45 @@
-    etamax    = cms.vdouble(  2.7,  2.7, 2.7,  2.4, 2.4,  3.0, 3.0, -1.0, -1.0,-1.0),
-    cutvalmin = cms.vdouble( -1.0, -1.0, 1.0,  0.0, 0.0, 0.02, 2.0, -1.0, 0.02,10.0),
-    cutvalmax = cms.vdouble( 0.90, 0.90,-1.0, -1.0,-1.0, 0.99,-1.0, 0.90, -1.0,-1.0),
+TMeras.TMUL2017.toModify(GoodJetsProducer,
```
If 2017 and 2018 really have all of the same values you can remove the 2018 block and instead write:
`(TMeras.TMUL2017 | TMeras.TMUL2018).toModify(GoodJetsProducer, ...`

aperloff commented on Utils/python/goodjetsproducer_cfi.py:
```diff
@@ -32,44 +34,45 @@
-    etamax    = cms.vdouble(  2.6,  2.6, 2.6,  2.6, 2.6,  2.7,  2.7, 2.7,  3.0, 3.0, -1.0, -1.0,-1.0),
-    cutvalmin = cms.vdouble( -1.0, -1.0, 1.0,  0.0, 0.0, -1.0, -1.0, 0.0, 0.02, 2.0, -1.0, 0.02,10.0),
-    cutvalmax = cms.vdouble( 0.90, 0.90,-1.0, -1.0,-1.0, 0.90, 0.99,-1.0, 0.99,-1.0, 0.90, -1.0,-1.0),
+TMeras.TMUL2018.toModify(GoodJetsPuppiProducer,
```
Same comment about what to do if `TMUL2017` and `TMUL2018` are the same.

aperloff commented on Utils/python/pdfweightproducer_cfi.py:
```diff
@@ -2,12 +2,7 @@
 PDFWeightProducer = cms.EDProducer("PDFWeightProducer",
     nScales = cms.uint32(9),
-    nPDFs = cms.uint32(100),
+    nPDFs = cms.uint32(102),
```
It's 102 for `TMUL2016` now?

kpedro88 replied:
Yes, all years now use the same NLO (but positive-definite) PDFs.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -22,10 +22,6 @@ class MCSampleHelper():
     """
 
     __energy_dict = {
-        "2010" : "7TeV",
```
Don't remove the year-energy mapping going back to 2010.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -34,7 +30,7 @@ class MCSampleHelper():
         "generators"    : ["(.pythia)([^_-]*)","(.powheg)([^_-]*)","(.madgraph)([^_-]*)","(.madspin)([^_-]*)","(.amcatnlo)([^_-]*)"],
         "tunes"         : ["(.Tune)([^_]*)"],
-        "added_info"    : ["(.PSweights)([^_]*)"],
+        "added_info"    : ["(.PSweights)([^_]*)","(.PSWeights)([^_]*)"],
```
I think the two psweights patterns be combined using `(.PS[W,w]eights)([^_]*)`. I tried it on https://regex101.com/ and it seems to work for the sample lines:
```
stuff_PSweights_stuff
stuff_PSWeights_stuff
```

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -120,7 +116,7 @@ class MCSampleValuesHelper():
         "TTJets" : {
             "CrossSection" : XSValues(
-                XS_13TeV=831.76, XSSource_13TeV="XSDB (NNLO)",
+                XS_13TeV=750.5, XSSource_13TeV="XSDB (unknown)",
```
`831.76` is the inclusive NNLO XS from the PDG (https://pdg.lbl.gov/2019/reviews/rpp2019-rev-top-quark.pdf). It was most likely put in by hand. Where did you get 750.5? I suspect that's not correct and that likely this one shouldn't be changed.

JHiltbrand replied:
Yeah, these numbers coming from the XSDB do not have much context or justification. When possible, I will stick to the highest order calculated values from PDG, etc.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
         "TTTo2L2Nu" : {
             "CrossSection" : XSValues(
-                XS_13TeV=88.29, XSSource_13TeV="XSDB (NNLO)",
+                XS_13TeV=687.1, XSSource_13TeV="XSDB (NLO)",
```
This is a big change and a step down from NNLO. Where did `687.1` come from? Does it take into account the BR to 2L2Nu?

aperloff replied:
Perhaps, if this is what's missing, we should use the BranchingRatio mechanism (i.e. https://github.com/TreeMaker/TreeMaker/pull/573/files#diff-8fd74e435000e63630ca4a7ff70eb8557d93718085f7e34fcf71f5001b275b99L1032) to make it more clear where the values are coming from.

JHiltbrand replied:
Again, the XSDB delivers no insight into where the 687.1. I see what happened with the Br part as well. Will amend with latest PDG numbers.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
-            "Correction" : CorrValues(
-                Corr_2017=0.993, CorrSource_2017="https://twiki.cern.ch/twiki/bin/viewauth/CMS/MCKnownIssues#WJetsToLNu_HT_and_DYJets_HT_LO_M",
-                Corr_2018=0.993, CorrSource_2018="Same as 2017",
+                XS_13TeV=687.1, XSSource_13TeV="XSDB (NLO)",
```
Again, looks like you're using an NLO and not the NNLO PDG value. Also, you appear to be missing the BR to all-hadronic (66.5% from https://pdg.lbl.gov/2018/listings/rpp2018-list-t-quark.pdf). Again, we could use the built-in BR mechanism to separate the cross sections and branching ratios so that this confusion doesn't happen again.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
-                XS_2016=1.329, XSSource_2016="XSDB (LO)",
-                XS_2017=1.074, XSSource_2017="XSDB (LO)",
-                XS_2018=1.085, XSSource_2018="GenXSecAnalyzer",
+                XS_13TeV=687.1, XSSource_13TeV="XSDB (NLO)",
```
Please review all of the TTJets samples so that I don't keep making the same comment.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
             "CrossSection" : XSValues(
-                XS_2017=34.71, XSSource_2017="GenXSecAnalyzer",
-                XS_2018=34.74, XSSource_2018="GenXSecAnalyzer",
+                XS_13TeV=1.502, XSSource_13TeV="XSDB (LO)",
```
Are the `TTGamma_SingleLept`, `TTGamma_Hadronic`, and `TTGamma_Dilep` cross sections the same for all years? They used to be different for each year. Also, are the samples `TTGamma_SingleLeptFromT` and `TTGamma_SingleLeptFromTbar` samples combined into the `TTGamma_SingleLept` sample? If so, why is it's cross section so much higher than the sum of the old ones.

JHiltbrand replied:
So far, I have not been able to find any evidence that cross sections will be different on a per-year basis. I do find [1], which discusses a uniform approach for MC generation for all three UL years, helping support the assumption that all three years are indeed treated equally.

From what I can tell, the top and antitop modes for single lepton are now just one.  As for the cross section discrepancy, another thing to investigate.

[1] https://cms-pdmv.gitbook.io/project/mccontact/info-for-mc-production-for-ultra-legacy-campaigns-2016-2017-2018

aperloff replied:
I also don't see the value of 1.502 in the XSDB for the search `DAS=TTGamma_Dilept && DAS=TuneCP5`. Just thought I'd mention that.

JHiltbrand replied:
I guess this [2] is what I have been going off of for Dilept in particular.

[2] https://cms-gen-dev.cern.ch/xsdb/?columns=67108800&currentPage=0&pageSize=10&searchQuery=DAS%3DTTGamma_Dilept_TuneCP5_13TeV-madgraph-pythia8

aperloff replied:
Okay, I see the value for TTGamma_Dilept in the XSDB now, but that still doesn't explain the XS change. Not unless the madgraph generation has changed.

JHiltbrand replied:
Hmm, when I specifically follow a Fall17 or Autumn18 TTGamma_Dilept sample _to_ XSDB [3] from DAS [4], it points to an entry with a cross section very close to the 1.5 value. For Summer16 samples with the TuneCUETPM8 tune, I do find the ~0.6 cross section. There are XSDB entries for Fall17 that have the smaller value reported and were part of a different chain, but otherwise have the same DAS name: compare top two rows [5].

The MadGraph version did change from 2.6.0 to 2.6.5.

[3] https://cmsweb.cern.ch/das/request?input=dataset%3D%2FTTGamma_Dilept_TuneCP5_13TeV-madgraph-pythia8%2FRunIIAutumn18MiniAOD-102X_upgrade2018_realistic_v15-v1%2FMINIAODSIM&instance=prod/global

[4] https://cms-gen-dev.cern.ch/xsdb/?searchQuery=DAS=TTGamma_Dilept_TuneCP5_13TeV-madgraph-pythia8

[5] https://cms-gen-dev.cern.ch/xsdb/?columns=67108863&currentPage=0&pageSize=10&searchQuery=DAS%3DTTGamma_Dilept_TuneCP5_

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
-                XS_2016=99.65, XSSource_2016="XSDB (LO)"
+        "WJetsToLNu" : {
+           "CrossSection" : XSValues(
+                 XS_13TeV=53870.0, XSSource_13TeV="XSDB (LO)",
```
1. Please use the PDG value for the `XS_13TeV` value as in https://github.com/TreeMaker/TreeMaker/pull/573/files#diff-8fd74e435000e63630ca4a7ff70eb8557d93718085f7e34fcf71f5001b275b99L414.
2. Please also mark down the LO XS per year.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
-                XS_2016=1710000.0, XSSource_2016="XSDB (LO)",
-                XS_2017=1547000.0, XSSource_2017="XSDB (LO)",
-                XS_2018=1557000.0, XSSource_2018="XSDB (LO)",
+                XS_13TeV=1264.0, XSSource_13TeV="XSDB (LO)",
```
Same for all three years? This is the last time I make this comment until I hear whether or not that's true in general.

kpedro88 replied:
I think all years should be the same now that the PDFs are the same (but it would be good to confirm this).

JHiltbrand replied:
See reply to TTGamma post.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
-                XS_2016=32060.0, XSSource_2016="XSDB (LO)",
-                XS_2017=29980.0, XSSource_2017="XSDB (LO)",
-                XS_2018=30140.0, XSSource_2018="XSDB (LO)",
+                XS_13TeV=335.5, XSSource_13TeV="XSDB (LO)",
```
Seems a lot lower than the EOY XS. Do you know why? Please double check this and the other WJetsToLNu_HT bins.

JHiltbrand replied:
I opened up a dialog in the GEN production hypernews [1]. Some reasons for the differences for WJets have been provided.

[1] https://hypernews.cern.ch/HyperNews/CMS/get/generators/5172.html

aperloff replied:
Please collect these explanations and make a list of them (short summaries) in https://github.com/TreeMaker/TreeMaker/issues/584. That way I can refer to that issue when I'm next reviewing this PR and not have to go clicking through tons of links.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
+                 XS_13TeV=106500000.0, XSSource_13TeV="XSDB (unknown)",
             ),
         },
         "QCD_Pt_50to80" : {
```
Do you not need the MuEnriched samples anymore?

JHiltbrand replied:
Yeah, when reviewing with @cmadrid1 the samples the Stealth/RPV group has been using, we do not see a need for them going forward.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
-                kFactor_2017=1.1374, kFactorSource_2017="XSDB NNLO/LO=6077.22/5343",
-                kFactor_2018=1.1421, kFactorSource_2018="XSDB NNLO/LO=6077.22/5321",
-            ),
-            "Correction" : CorrValues(
```
We'll probably want to double-check after the initial production that these correction factors are no longer needed.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,669 +247,501 @@ class MCSampleValuesHelper():
         "DYJetsToLL_M-50" : {
             "CrossSection" : XSValues(
-                XS_13TeV=6077.22, XSSource_13TeV="XSDB (NNLO)",
+                XS_13TeV=5398, XSSource_13TeV="XSDB (LO)",
```
Use the latest NNLO number.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -935,62 +763,52 @@ class MCSampleValuesHelper():
+                XS_13TeV=41.39, XSSource_13TeV="GenXSecAnalyzer",
             ),
         },
+        #TODO
```
Please remove this comment and all of the other `#TODO` comments.

kpedro88 replied:
or clarify *what* to do

JHiltbrand replied:
Cleaned up, keeping track in the new issue todo list.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -999,141 +817,164 @@ class MCSampleValuesHelper():
-                XS_2016=38.06, XSSource_2016="XSDB (NLO)",
-                XS_2017=34.97, XSSource_2017="XSDB (NLO)",
-                XS_2018=34.97, XSSource_2018="GenXSecAnalyzer",
+                XS_13TeV=32.51, XSSource_13TeV="XSDB (NLO)",
```
Do you know why the UL `ST_tW` cross sections are systematically lower than any of the EOY cross sections?

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -999,141 +817,164 @@ class MCSampleValuesHelper():
         "WW" : {
             "CrossSection" : XSValues(
-                XS_13TeV=51.723, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#Diboson, WW>4q NNLO",
+                XS_13TeV=75.95, XSSource_13TeV="XSDB (unknown)",
```
This doesn't seem right. WW is a well measured process. You should use the NNLO number unless you have some other compelling reason.

JHiltbrand replied:
Indeed, using the NNLO is better. As an aside, I see the number for WW -> 4Q had been used here, whereas I would think we want the inclusive number for "WW", or am I missing something ?

aperloff replied:
The twiki doesn't have the inclusive cross section on it. The source they point to (https://arxiv.org/pdf/1408.5243.pdf) does, but I can't quite figure out how they got the numbers listed on the twiki. Maybe a more detailed reading would help.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -999,141 +817,164 @@ class MCSampleValuesHelper():
         "WZ" : {
             "CrossSection" : XSValues(
-                XS_13TeV=47.13, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#Diboson, inclusive NLO from MCFM",
+                XS_13TeV=27.59, XSSource_13TeV="XSDB (unknown)",
```
How is this XS 40% lower than before?

JHiltbrand replied:
Will stick to known-order cross section.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -999,141 +817,164 @@ class MCSampleValuesHelper():
         "ZZ" : {
             "CrossSection" : XSValues(
-                XS_13TeV=16.523, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#Diboson, inclusive NLO from MCFM",
+                XS_13TeV=12.17, XSSource_13TeV="XSDB (unknown)",
```
Why is this one 30% lower?

JHiltbrand replied:
Will stick to know-order cross section.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -999,141 +817,164 @@ class MCSampleValuesHelper():
         "WWTo2L2Nu" : {
             "CrossSection" : XSValues(
-                XS_13TeV=12.178, XSSource_13TeV="https://twiki.cern.ch/twiki/bin/viewauth/CMS/SummaryTable1G25ns#Diboson, WW>2l2v NNLO",
+                XS_13TeV=11.09, XSSource_13TeV="XSDB (NLO)",
```
Use the NNLO number when available.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -999,141 +817,164 @@ class MCSampleValuesHelper():
-                XS_2016=12.68,  XSSource_2016="XSDB (LO)",
-                XS_2017=12.93, XSSource_2017="XSDB (LO)",
-                XS_2018=12.95, XSSource_2018="GenXSecAnalyzer",
+                XS_13TeV=19.77,  XSSource_13TeV="GenXSecAnalyzer",
```
Any idea why this one is so much higher than before?

JHiltbrand replied:
Hmm, I don't see anything obvious. I reran the GenXSecAnalyzer and reobtained the same result.

aperloff replied:
Take a look at the gen fragment for either the 2017 or 2018 sample and then compare it to the gen fragment for the new UL sample. Is it only the tune that changed? If that's the case, then you should ask the GEN POG if this change might be expected for this process.

JHiltbrand replied:
Yeah, sounds good.

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -999,141 +817,164 @@ class MCSampleValuesHelper():
-                XS_2016=0.5644, XSSource_2016="GenXSecAnalyzer",
-                XS_2017=0.6008, XSSource_2017="GenXSecAnalyzer",
-                XS_2018=0.6008, XSSource_2018="GenXSecAnalyzer",
+                XS_13TeV=0.9738, XSSource_13TeV="XSDB (NLO)",
```
Seems a lot higher than before. Is that simply by going from LO to NLO?

JHiltbrand replied:
See hypernews thread [1] for some insight on this ZZTo2L2Nu sample.

[1] https://hypernews.cern.ch/HyperNews/CMS/get/generators/5172.html

aperloff reviewed (commented):
In addition to the review comments, I have some comments that don't fit in any specific file.
1. You have removed all of the `Production/test/condorSub/dict_*.py` files for data. Do you plan to add those back in?
2. Why did you add the JEC SQLite files for both the combined and individual eras for data? This goes for Summer19UL17 and Summer19UL18.

Commit:
remove pre-UL golden JS, remove per-major-run JEC/JER---not needed

Commit:
address some comments

Commit:
consolidate tth into tt

kpedro88 commented on Production/test/condorSub/dict_Summer20UL16APV_tt.py:
```diff
@@ -1,6 +1,19 @@
 flist = {
-    "scenario": "Summer20UL16APV",
-    "samples": [
-        ['Summer20UL16APV.TTToHadronic_TuneCP5_13TeV-powheg-pythia8'],
```
seems like you lost `TTToHadronic` in all of these

JHiltbrand replied:
Based on a comment from Alexx, anything with tt + X now lives in the `_tt.py` while the three decay modes for "pure" tt live in `_ttbar.py`.

kpedro88 replied:
Ah okay, I had missed this point.

Commit:
sort out DYJets for 16APV

Commit:
use sig scenario for signals

Commit:
add fastsim scenarios

Commit:
removed unspecified todos, explained in GitHub issue

Commit:
make some updates to MCSampleValues.py

JHiltbrand commented:
Dear @aperloff and @kpedro88,

My latest commit (57188f1) has several changes to the cross sections in `MCSampleValues.py` for review. Here I try to explain the changes:

#### TTJets and TT ####
 - switched to the known NNLO value and then applied the branching fraction for the respective channel, based on PDG.

#### WJetsToLNu ####
 - using the NNLO value
 - added a k factor for the HT-binned samples, based on the availble LO inclusive sample

#### DYJetsToLL ####
 - use the known NNLO value for the inclusive sample
 - added a k factor the HT-binned samples based on the LO inclusive sample

#### ZJetsToNuNu ####
 - updated the k factor based on DYJetsToLL information

#### ST ####
 - updated numbers to NLO based on twiki page reference in source variable

#### Diboson ####
 - for inclusive used the known NNLO,NLO values

#### TTWJets ####
 - added cross section and branching ratio information based on yellow report

#### ttH ####
 - updated values based on yellow report

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,744 +251,497 @@ class MCSampleValuesHelper():
         "TTTo2L2Nu" : {
             "CrossSection" : XSValues(
-                XS_13TeV=88.29, XSSource_13TeV="XSDB (NNLO)",
+                XS_13TeV=87.34, XSSource_13TeV="PDG XS - https://pdg.lbl.gov/2021/reviews/rpp2020-rev-top-quark.pdf",
```
Did you multiply in by hand the branching fraction? If so, why not use the branching fraction mechanism built into the MCSample/MCSampleHelper class?

JHiltbrand replied:
Ah, implemented branching fraction mechanism like in other places 33f4d81

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,744 +251,497 @@ class MCSampleValuesHelper():
-                XS_13TeV=377.96, XSSource_13TeV="XSDB (NNLO)",
+        "TTTo2L2Nu_mtop166p5" : {
+           "CrossSection" : XSValues(
+                 XS_13TeV=811.4, XSSource_13TeV="XSDB (NLO)",
```
How is this cross section, `811.4`, so many times higher than the `TTTo2L2Nu` cross section of `87.34`?

JHiltbrand replied:
Made the top mass variations behave like the nominal samples---by using branching fraction. Also implemented a k factor. 33f4d81

aperloff commented on WeightProducer/python/MCSampleValues.py:
```diff
@@ -251,744 +251,497 @@ class MCSampleValuesHelper():
-                XS_13TeV=365.34, XSSource_13TeV="XSDB (NNLO)",
+        "TTTo2L2Nu_mtop169p5" : {
+           "CrossSection" : XSValues(
+                 XS_13TeV=746.2, XSSource_13TeV="XSDB (NLO)",
```
Same comment as above

Commit:
fix up top mass variation cross sections

Commit:
add 2016 post VFP b tag WP, add triggers from Nadja
